nohup: ignoring input
original problem has 11923 variables (11923 bin, 0 int, 0 impl, 0 cont) and 15151 constraints
Traceback (most recent call last):
  File "/home/liudefen/projects/ML4LocalBranch_extend/primal_heur_localbranch.py", line 388, in heurexec
    self.agent_k, self.optim_k, R = self.update_agent(agent_k, self.optim_k)
  File "/home/liudefen/projects/ML4LocalBranch_extend/primal_heur_localbranch.py", line 601, in update_agent
    policy_losses = torch.cat(policy_losses).sum()
RuntimeError: There were no tensor arguments to this function (e.g., you passed an empty list of Tensors), but no fallback function is registered for schema aten::_cat.  This usually means that this function requires a non-empty list of Tensors.  Available functions are [CPU, CUDA, QuantizedCPU, BackendSelect, Named, AutogradOther, AutogradCPU, AutogradCUDA, AutogradXLA, AutogradPrivateUse1, AutogradPrivateUse2, AutogradPrivateUse3, Tracer, Autocast, Batched, VmapMode].

CPU: registered at /pytorch/build/aten/src/ATen/CPUType.cpp:2127 [kernel]
CUDA: registered at /pytorch/build/aten/src/ATen/CUDAType.cpp:2983 [kernel]
QuantizedCPU: registered at /pytorch/build/aten/src/ATen/QuantizedCPUType.cpp:297 [kernel]
BackendSelect: fallthrough registered at /pytorch/aten/src/ATen/core/BackendSelectFallbackKernel.cpp:3 [backend fallback]
Named: registered at /pytorch/aten/src/ATen/core/NamedRegistrations.cpp:7 [backend fallback]
AutogradOther: registered at /pytorch/torch/csrc/autograd/generated/VariableType_2.cpp:8078 [autograd kernel]
AutogradCPU: registered at /pytorch/torch/csrc/autograd/generated/VariableType_2.cpp:8078 [autograd kernel]
AutogradCUDA: registered at /pytorch/torch/csrc/autograd/generated/VariableType_2.cpp:8078 [autograd kernel]
AutogradXLA: registered at /pytorch/torch/csrc/autograd/generated/VariableType_2.cpp:8078 [autograd kernel]
AutogradPrivateUse1: registered at /pytorch/torch/csrc/autograd/generated/VariableType_2.cpp:8078 [autograd kernel]
AutogradPrivateUse2: registered at /pytorch/torch/csrc/autograd/generated/VariableType_2.cpp:8078 [autograd kernel]
AutogradPrivateUse3: registered at /pytorch/torch/csrc/autograd/generated/VariableType_2.cpp:8078 [autograd kernel]
Tracer: registered at /pytorch/torch/csrc/autograd/generated/TraceType_2.cpp:9654 [kernel]
Autocast: registered at /pytorch/aten/src/ATen/autocast_mode.cpp:258 [kernel]
Batched: registered at /pytorch/aten/src/ATen/BatchingRegistrations.cpp:511 [backend fallback]
VmapMode: fallthrough registered at /pytorch/aten/src/ATen/VmapModeRegistrations.cpp:33 [backend fallback]

Exception ignored in: 'pyscipopt.scip.PyHeurExec'
Traceback (most recent call last):
  File "/home/liudefen/projects/ML4LocalBranch_extend/primal_heur_localbranch.py", line 388, in heurexec
    self.agent_k, self.optim_k, R = self.update_agent(agent_k, self.optim_k)
  File "/home/liudefen/projects/ML4LocalBranch_extend/primal_heur_localbranch.py", line 601, in update_agent
    policy_losses = torch.cat(policy_losses).sum()
RuntimeError: There were no tensor arguments to this function (e.g., you passed an empty list of Tensors), but no fallback function is registered for schema aten::_cat.  This usually means that this function requires a non-empty list of Tensors.  Available functions are [CPU, CUDA, QuantizedCPU, BackendSelect, Named, AutogradOther, AutogradCPU, AutogradCUDA, AutogradXLA, AutogradPrivateUse1, AutogradPrivateUse2, AutogradPrivateUse3, Tracer, Autocast, Batched, VmapMode].

CPU: registered at /pytorch/build/aten/src/ATen/CPUType.cpp:2127 [kernel]
CUDA: registered at /pytorch/build/aten/src/ATen/CUDAType.cpp:2983 [kernel]
QuantizedCPU: registered at /pytorch/build/aten/src/ATen/QuantizedCPUType.cpp:297 [kernel]
BackendSelect: fallthrough registered at /pytorch/aten/src/ATen/core/BackendSelectFallbackKernel.cpp:3 [backend fallback]
Named: registered at /pytorch/aten/src/ATen/core/NamedRegistrations.cpp:7 [backend fallback]
AutogradOther: registered at /pytorch/torch/csrc/autograd/generated/VariableType_2.cpp:8078 [autograd kernel]
AutogradCPU: registered at /pytorch/torch/csrc/autograd/generated/VariableType_2.cpp:8078 [autograd kernel]
AutogradCUDA: registered at /pytorch/torch/csrc/autograd/generated/VariableType_2.cpp:8078 [autograd kernel]
AutogradXLA: registered at /pytorch/torch/csrc/autograd/generated/VariableType_2.cpp:8078 [autograd kernel]
AutogradPrivateUse1: registered at /pytorch/torch/csrc/autograd/generated/VariableType_2.cpp:8078 [autograd kernel]
AutogradPrivateUse2: registered at /pytorch/torch/csrc/autograd/generated/VariableType_2.cpp:8078 [autograd kernel]
AutogradPrivateUse3: registered at /pytorch/torch/csrc/autograd/generated/VariableType_2.cpp:8078 [autograd kernel]
Tracer: registered at /pytorch/torch/csrc/autograd/generated/TraceType_2.cpp:9654 [kernel]
Autocast: registered at /pytorch/aten/src/ATen/autocast_mode.cpp:258 [kernel]
Batched: registered at /pytorch/aten/src/ATen/BatchingRegistrations.cpp:511 [backend fallback]
VmapMode: fallthrough registered at /pytorch/aten/src/ATen/VmapModeRegistrations.cpp:33 [backend fallback]

./result/saved_models/trained_params_mean_generalized_independentset-small_symmetric_firstsol_k_prime.pth
./result/saved_models/rl/reinforce/setcovering/checkpoint_trained_reward3_simplepolicy_rl4lb_reinforce_trainset_setcovering-small_lr0.01_epochs7.pth
generalized_independentset-small
firstsol
seed: 0
cpu
instance:  0
create a new SCIP model
The initial incumbent of generalized_independentset-80_transformed is successfully added to MIP model
generalized_independentset-80_transformed
N of variables: 11923
N of binary vars: 11923
N of constraints: 15151
Initial obj of MIP: 0.0
* Model status: unknown
* Model status: nodelimit
* Solve stage: 9
* LP status: 1
* number of sol :  26
GNN prediction:  0.14815136790275574
GNN k_0:  42.0
LB heuristic is starting..
LB heuristic starts at 0.151344 s
No. of LB call:  1
LB initial objective: -4500.0
The init sol of original MIP is feasible
LB round: 1 Solving time: 2.0130 Best Obj: -4500.0000 n_sols_subMIP: 2 K: 42 t_node: 2.0 self.div: 0 LB Status: timelimit
K_final: 42 div_final: 0
Traceback (most recent call last):
  File "evaluation_scip_lb_regression_rl.py", line 119, in <module>
    scip_as_baseline.execute_heuristic_baseline(
  File "/home/liudefen/projects/ML4LocalBranch_extend/execute_heuristics.py", line 371, in execute_heuristic_baseline
    self.setup_heuristic_per_instance(MIP_model=MIP_model,
  File "/home/liudefen/projects/ML4LocalBranch_extend/execute_heuristics.py", line 290, in setup_heuristic_per_instance
    data = self.execute_heuristic_per_instance(MIP_model, incumbent, node_time_limit,
  File "/home/liudefen/projects/ML4LocalBranch_extend/execute_heuristics.py", line 2224, in execute_heuristic_per_instance
    MIP_model_copy2.optimize()
  File "src/pyscipopt/scip.pyx", line 3158, in pyscipopt.scip.Model.optimize
  File "src/pyscipopt/scip.pyx", line 231, in pyscipopt.scip.PY_SCIP_CALL
Exception: SCIP: unspecified error!
